pkg/resourceinterpreter/default/native/retain_test.go

```diff
@@ -113,6 +113,10 @@ func Test_retainK8sWorkloadReplicas(t *testing.T) {
 func Test_retainSecretServiceAccountToken(t *testing.T) {
 	createSecret := func(secretType corev1.SecretType, uuid, key, value string) *unstructured.Unstructured {
 		ret, _ := helper.ToUnstructured(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{corev1.ServiceAccountUIDKey: uuid},
+			},
+			Data: map[string][]byte{key: []byte(value)},
```

Comment:
Can you remind me why ObjectMeta and Data are required here?

Reply:
Function `retainSecretServiceAccountToken` is used to retain the `uid` and `data` of observed secret when its type is SA. Therefore the input secret of its test case should contain `ObjectMeta`(for uid) and `Data`. The previous implementations didn't actually test the ability to retain `uid` and `data`, because `ObjectMeta` and `Data` weren't used when creating test case secrets

Reply:
Got it, thanks.

@XiShanYongYe-Chang I'm not sure if you have to revisit this test at #4940 after this.